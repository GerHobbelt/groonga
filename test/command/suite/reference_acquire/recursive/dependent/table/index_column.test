#$GRN_ENABLE_REFERENCE_COUNT=yes

table_create Tags TABLE_HASH_KEY ShortText

table_create Memos TABLE_HASH_KEY ShortText
column_create Memos content COLUMN_SCALAR Text

table_create Users TABLE_HASH_KEY ShortText
column_create Users memos COLUMN_VECTOR Memos

column_create Memos users_memos COLUMN_INDEX Users memos

table_create Terms TABLE_PAT_KEY ShortText --default_tokenizer TokenBigram
column_create Terms memos_content COLUMN_INDEX|WITH_POSITION Memos content
column_create Terms tags_key COLUMN_INDEX|WITH_POSITION Tags _key

log_level --level dump
#@add-important-log-levels dump
#@add-ignore-log-pattern /\A\[io\]/
reference_acquire --target_name Memos --recursive dependent
select Memos
reference_release --target_name Memos --recursive dependent
