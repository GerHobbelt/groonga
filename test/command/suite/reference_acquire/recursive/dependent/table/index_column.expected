table_create Tags TABLE_HASH_KEY ShortText
[[0,0.0,0.0],true]
table_create Memos TABLE_HASH_KEY ShortText
[[0,0.0,0.0],true]
column_create Memos content COLUMN_SCALAR Text
[[0,0.0,0.0],true]
table_create Users TABLE_HASH_KEY ShortText
[[0,0.0,0.0],true]
column_create Users memos COLUMN_VECTOR Memos
[[0,0.0,0.0],true]
column_create Memos users_memos COLUMN_INDEX Users memos
[[0,0.0,0.0],true]
table_create Terms TABLE_PAT_KEY ShortText --default_tokenizer TokenBigram
[[0,0.0,0.0],true]
column_create Terms memos_content COLUMN_INDEX|WITH_POSITION Memos content
[[0,0.0,0.0],true]
column_create Terms tags_key COLUMN_INDEX|WITH_POSITION Tags _key
[[0,0.0,0.0],true]
log_level --level dump
[[0,0.0,0.0],true]
reference_acquire --target_name Memos --recursive dependent
[[0,0.0,0.0],true]
#|-| [obj][open] <257>(<Memos>):<48>(<table:hash_key>)
#|-| [obj][open] <258>(<Memos.content>):<65>(<column:var_size>)
#|-| [obj][open] <262>(<Terms>):<49>(<table:pat_key>)
#|-| [obj][open] <263>(<Terms.memos_content>):<72>(<column:index>)
#|-| [obj][open] <264>(<Terms.tags_key>):<72>(<column:index>)
#|-| [obj][close] <264>(<Terms.tags_key>):<72>(<column:index>)
#|-| [obj][open] <261>(<Memos.users_memos>):<72>(<column:index>)
#|-| [obj][open] <259>(<Users>):<48>(<table:hash_key>)
#|-| [obj][open] <260>(<Users.memos>):<65>(<column:var_size>)
select Memos
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        0
      ],
      [
        [
          "_id",
          "UInt32"
        ],
        [
          "_key",
          "ShortText"
        ],
        [
          "content",
          "Text"
        ],
        [
          "users_memos",
          "UInt32"
        ]
      ]
    ]
  ]
]
reference_release --target_name Memos --recursive dependent
[[0,0.0,0.0],true]
#|-| [obj][open] <264>(<Terms.tags_key>):<72>(<column:index>)
#|-| [obj][close] <264>(<Terms.tags_key>):<72>(<column:index>)
#|-| [obj][close] <263>(<Terms.memos_content>):<72>(<column:index>)
#|-| [obj][close] <262>(<Terms>):<49>(<table:pat_key>)
#|-| [obj][close] <258>(<Memos.content>):<65>(<column:var_size>)
#|-| [obj][close] <260>(<Users.memos>):<65>(<column:var_size>)
#|-| [obj][close] <259>(<Users>):<48>(<table:hash_key>)
#|-| [obj][close] <261>(<Memos.users_memos>):<72>(<column:index>)
#|-| [obj][close] <257>(<Memos>):<48>(<table:hash_key>)
